x-pack/plugins/apm/common/rules/schema.ts

```diff
@@ -9,6 +9,16 @@ import { schema, TypeOf } from '@kbn/config-schema';
 import { ML_ANOMALY_SEVERITY } from '@kbn/ml-anomaly-utils/anomaly_severity';
 import { AggregationType, ApmRuleType } from './apm_rule_types';
 
+export const searchConfigurationSchema = schema.object({
+  query: schema.object({
+    query: schema.oneOf([
+      schema.string(),
+      schema.recordOf(schema.string(), schema.any()),
+    ]),
+    language: schema.string(),
+  }),
+});
```

Comment:
Is this specific to APM? I thought the idea was to share the same naming/structure across observability?

Reply:
We are not using unified search bar/searchConfiguration in other observability rules yet. I only implemented in APM rules to keep PR scope limited.

ES rule (and maybe some other rules) uses `searchConfiguration` and unified search bar. The idea is that when we look at the code, we find similar pattern and UI gives same UX with unified search bar.  We are going to implement the same in Threshold rule.